pkg/apis/k8s/v1alpha1/nginxingresscontroller_types.go

```diff
@@ -86,6 +90,12 @@ type NginxIngressControllerSpec struct {
 	// – only one replica will report status.
 	// +kubebuilder:validation:Optional
 	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	IngressLink string `json:"ingressLink,omitempty"`
```

Comment:
better to move this under `ReportIngressStatus` as we do in the helm chart, as IngressLink resource is used as the source of the IP address to report. 

also, because of this check https://github.com/nginxinc/kubernetes-ingress/blob/master/cmd/nginx-ingress/main.go#L253 , can we make sure operator doesn't set both flags? For example, one of them can take precedence, like `ingressLink`